robots-module/src/main/java/orwell/tank/inputs/ConnectedToProxy.java

```diff
@@ -0,0 +1,39 @@
+package orwell.tank.inputs;
+
+import orwell.tank.EnumConnectionState;
+import orwell.tank.IInputVisitor;
+import orwell.tank.Tank;
+import orwell.tank.elements.DisplayScreen;
+import orwell.tank.elements.IDrivingTracks;
+import orwell.tank.elements.RfidFlagSensor;
+import orwell.tank.elements.SoundSpeaker;
+
+/**
+ * Created by Michaël Ludmann on 6/15/15.
+ */
+public class ConnectedToProxy implements IInputVisitor {
+    @Override
+    public void visit(Tank tank) {
+        tank.setConnectionState(EnumConnectionState.CONNECTED);
```

Comment:
This seems very complicated to notify the tank that the connection is established. Why is it not pointing to the object handling the connection directly ?